crawler/src/main/java/com/brandwatch/minibcr/crawler/service/reddit/RedditCrawler.java

```diff
@@ -0,0 +1,72 @@
+package com.brandwatch.minibcr.crawler.service.reddit;
+
+import com.brandwatch.minibcr.crawler.model.Resource;
+import com.brandwatch.minibcr.crawler.model.reddit.SubReddit;
+import com.brandwatch.minibcr.crawler.service.Crawler;
+import com.brandwatch.minibcr.crawler.service.reddit.auth.RedditAuthenticator;
+import com.brandwatch.minibcr.crawler.service.reddit.auth.RedditClient;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Collections;
+
+@Service
+public class RedditCrawler implements Crawler {
+
+    private static final String SUBREDDIT_NAME = "phone";
+    private static final String REDDIT_URL = "https://oauth.reddit.com/r/%s/new?limit=100";
+
+    private final RedditAuthenticator authenticator;
+    private final RedditClient redditClient;
+
+    private final RestTemplate restTemplate;
+
+    private final Producer producer;
+
+
+    public RedditCrawler(
+            RedditAuthenticator authenticator,
+            RedditClient redditClient,
+            RestTemplate restTemplate,
+            Producer producer
+    ) {
+        this.authenticator = authenticator;
+        this.redditClient = redditClient;
+        this.restTemplate = restTemplate;
+        this.producer = producer;
+    }
+
+    @Override
+    public void crawl() {
```

Comment:
Should we schedule this to run every 5 mins for example? Or what's your plan for executing this method?

Reply:
Added schedule configurations